src/hooks/table/use-select.tsx

```diff
@@ -0,0 +1,65 @@
+import { useCallback, useState, ChangeEvent } from 'react'
+
+interface Item {
+  _id: string
+}
+
+interface SelectHook {
+  selected: string[]
+  isSelected: (id: string) => boolean
+  clearSelected: () => void
+  createSelectAllHandler: (
+    items: Item[]
+  ) => (e: ChangeEvent<HTMLInputElement>) => void
+  handleSelectClick: (id: string) => void
+}
+
+const useSelect = ({
+  initialSelect = []
+}: {
+  initialSelect?: string[]
+}): SelectHook => {
```

Comment:
The more readable way is to type things here is to create separate interfaces for arguments and return type. I suggest the following:

```jsx
interface UseSelectInput {
    // ... your fields
}

interface UseSelectOutput {
    // ... your fields
}

const useSelect = ({ initialSelect = [] }: UseSelectInput): UseSelectOutput // ... other code
```

You can find similar approach inside `useForm` or `useFilterQuery` hooks

Reply:
Thanck uoy for your advice, but I think that in this case it is not necessary and will only make readability more difficult

Reply:
It definitely will improve readability in case adding more props in the future. And by the way it is much cleaner way to declare types for function